### PR TITLE
API

### PR DESCRIPTION
**Pull request by robinandeer**
This turned out really nice ^^

I created a new API class which inherits from the base `Store` class

Before closing #148 we should merge #146, rebase this branch and update the CLI :smile:

### CODE REVIEW

File: chanjo/store/api.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+        Returns:
+            List[Sample]: samples objects from the database
+        """
+        query = self.query(Sample).order_by(Sample.sample_id)
+        query = filter_samples(query, group_id=group_id, sample_ids=sample_ids)
+        return query
+
+    def mean(self, *samples):
+        r"""Calculate mean values for all metrics on a per sample basis.
```
Comment (moonso) [inline chanjo/store/api.py]: What about this 'r' in the beginning?

File: chanjo/store/api.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+        if per == 'exon':
+            results = results.group_by(Exon.exon_id)
+            exon_groups = group_by_field(results, name='exon_id')
+            return exon_groups
+        else:
+            data = {metric: value for exon_id, metric, value in results}
+            return data
+
+    def gene(self, *gene_ids):
```
Comment (moonso) [inline chanjo/store/api.py]: Super that you already implemented this!

File: chanjo/store/api.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+                              self.weighted_average)
+                       .join(ExonStatistic.sample, ExonStatistic.exon)
+                       .group_by(Sample.sample_id, ExonStatistic.metric))
+        if samples:
+            results = results.filter(Sample.sample_id.in_(samples))
+
+        sample_groups = group_by_field(results, name='sample_id')
+        return sample_groups
+
+    def region(self, chromosome, start, end, sample_id=None, per=None):
```
Comment (moonso) [inline chanjo/store/api.py]: Could it be possible to use a region id here to? As an alternative to coordinates?

File: chanjo/store/api.py [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+                            Exon.exon_id,
+                            ExonStatistic.value)
+                     .join(ExonStatistic.exon, ExonStatistic.sample)
+                     .filter(ExonStatistic.metric == completeness,
+                             ExonStatistic.value < threshold))
+        query = filter_samples(query, group_id=group_id, sample_ids=sample_ids)
+        return query
+
+    def gene_panel(self, gene_ids, group_id=None, sample_ids=None):
+        """Report metrics for a panel of genes.
```
Comment (moonso) [inline chanjo/store/api.py]: Could you specify what metrics?

**Comment by moonso** [comment]
Nice work!!

Can you do the rebase? Next feature is to make use of this API for the command line tool?

File: chanjo/store/api.py (hunk 4 above)
Comment (robinandeer) [inline chanjo/store/api.py]: I can be a little more specific but it depends on what you ran Sambamba with!

File: chanjo/store/api.py (hunk 3 above)
Comment (robinandeer) [inline chanjo/store/api.py]: added another method and handled it in the CLI

**Comment by robinandeer** [comment]
I think we could push this to pip now under a beta flag, no?

**Pull request merged**